src/handlePress.js

```diff
@@ -40,6 +45,9 @@ export default (
     const { querySerializer: serializer } = getOptions()
     let action = isAction(to) ? to : pathToAction(url, routesMap, serializer)
     action = dispatchRedirect ? redirect(action) : action
+    if (action.type === NOT_FOUND) {
```

Comment:
This actually can't be how we deal with this. NOT_FOUND will never be able to be dispatched. People use NOT_FOUND to show 404 pages and the like when actual paths fail to convert. 

Instead short-circuit at the top of the file with url === '#'. We're essentially treating this as an intentional NOOP from the user.

Reply:
Oh right. I also think it's worth to add a unit test too.

Reply:
go for it